Revisit our node minimal version

I just tried running node 12.9, our minimal version, and by chance, tried the downloadOsmNetworkData task and got `Error: Cannot find module 'node:stream'`. Node 16 works fine. We should maybe update our minimum requirement?

--- a/.github/workflows/transition.yml
+++ b/.github/workflows/transition.yml
@@ -35,6 +35,33 @@ jobs:
       run: yarn test
     - name: UI Test
       run: yarn test:ui
+
+  # Because of https://github.com/chairemobilite/transition/issues/531, version 18 needs an additional environment variable. When fixed, we can use a version matrix instead
+  build-and-test-18:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18.x]
+    env:
+      PROJECT_CONFIG: ${{ github.workspace }}/examples/config.js
+    steps:
+    - uses: actions/checkout@v3
+    - name: copy env file
+      run: cp .env.example .env
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Install
+      run: yarn
+    - name: Compile
+      run: yarn compile
+    - name: Build Client bundle
+      run: yarn cross-env NODE_OPTIONS=--openssl-legacy-provider yarn build:prod
+    - name: Unit Test
+      run: yarn test
+    - name: UI Test
+      run: yarn test:ui
   
   code-lint:
     runs-on: ubuntu-latest

--- a/README.md
+++ b/README.md
@@ -93,6 +93,8 @@ Run `yarn build:dev` or `yarn build:prod` to create the html client application 
 
 The `prod` version is minified, while the `dev` version lhas greater size but allows to more easily debug the application.
 
+*Note: If running with node version 18, webpack may return an error. In that case, run `export NODE_OPTIONS=--openssl-legacy-provider` before.*
+
 ### Start the json2capnp cache server
 
 *Optional*

--- a/packages/chaire-lib-backend/package.json
+++ b/packages/chaire-lib-backend/package.json
@@ -1,6 +1,6 @@
 {
     "engines": {
-        "node": ">=12.9.0"
+        "node": ">=16.0.0"
     },
     "name": "chaire-lib-backend",
     "version": "0.2.2",

--- a/packages/chaire-lib-common/package.json
+++ b/packages/chaire-lib-common/package.json
@@ -1,6 +1,6 @@
 {
   "engines": {
-    "node": ">=12.9.0"
+    "node": ">=16.0.0"
   },
   "name": "chaire-lib-common",
   "version": "0.2.2",

--- a/packages/chaire-lib-frontend/package.json
+++ b/packages/chaire-lib-frontend/package.json
@@ -1,6 +1,6 @@
 {
   "engines": {
-    "node": ">=12.9.0"
+    "node": ">=16.0.0"
   },
   "name": "chaire-lib-frontend",
   "version": "0.2.2",

--- a/packages/transition-backend/package.json
+++ b/packages/transition-backend/package.json
@@ -1,6 +1,6 @@
 {
     "engines": {
-        "node": ">=12.9.0"
+        "node": ">=16.0.0"
     },
     "name": "transition-backend",
     "version": "0.2.0-2",

--- a/packages/transition-common/package.json
+++ b/packages/transition-common/package.json
@@ -1,6 +1,6 @@
 {
   "engines": {
-    "node": ">=12.9.0"
+    "node": ">=16.0.0"
   },
   "name": "transition-common",
   "version": "0.2.0-2",

--- a/packages/transition-frontend/package.json
+++ b/packages/transition-frontend/package.json
@@ -1,6 +1,6 @@
 {
   "engines": {
-    "node": ">=12.9.0"
+    "node": ">=16.0.0"
   },
   "name": "transition-frontend",
   "version": "0.2.0-2",

